mail/resourceAvailability.go

```diff
@@ -0,0 +1,117 @@
+package mail
+
+import (
+	"fmt"
+	"github.com/PPSKSY-Cluster/backend/db"
+	"github.com/go-co-op/gocron"
+	"log"
+	"os"
+	"sort"
+	"strconv"
+	"time"
+)
+
+var s = gocron.NewScheduler(time.Local)
+
+var check = func() {
+	if err := checkNotifications(); err != nil {
+		log.Println(err.Error())
+	}
+}
+
+func InitSchedule() error {
+
+	every, err := strconv.Atoi(os.Getenv("NOTIFICATION_INTERVAL"))
+	if err != nil {
+		return err
+	}
+
+	if s.Len() == 0 {
+		_, err := s.Every(every).
+			Second().
+			Do(check)
+		if err != nil {
+			return err
+		}
+
+		s.StartAsync()
+
+		fmt.Println("Initialized schedule")
+	}
+
+	return nil
+}
+
+func checkNotifications() error {
+	notifications, err := db.GetAllNotifications()
+	if err != nil {
+		return err
+	}
+
+	for _, n := range notifications {
+		if err := checkAvailability(n); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func checkAvailability(notification db.ReservationNotification) error {
+	user, err := db.GetUserById(notification.UserID)
+	if err != nil {
+		return err
+	}
+
+	cluster, err := db.GetCResourceById(notification.ClusterID)
+	if err != nil {
+		return err
+	}
+
+	reservations, err := db.GetReservationsByClusterId(notification.ClusterID)
+	if err != nil {
+		return err
+	}
+
+	clusterReservations := make(map[int64]int64) //Maps startTime to nodes used
+	var keyList []int64
+	for _, r := range reservations {
+		for start := r.StartTime; start < time.Now().AddDate(0, 3, 0).Unix(); start += 86400 { //1 Day = 86400 seconds
```

Comment:
I suppose 3 moths is the max time we look into the future for open time slots ? I think we should make this easy to find for people to configure. For now a const at the top of the file would be sufficient, later we should probably add a config file of some sort (some pairs from .env also aren't secrets and could go in a regular config)

Reply:
Yeah, thats a good idea